### PR TITLE
clio-40271 create datepicker filters

### PR DESCRIPTION
**Pull request by JeffMarvin**

### CODE REVIEW

File: themis_components/thDatePicker/examples/1 - Default Date Format/coffee.coffee [hunk 1]
```diff
@@ -1,5 +1,3 @@
 angular.module('thDemo', ['ThemisComponents'])
   .controller 'DemoCtrl', ->
-    @date = moment '2016-03-05'
```
Comment (JeffMarvin) [inline themis_components/thDatePicker/examples/1 - Default Date Format/coffee.coffee]: Part of this ticket changes the thDatePicker component to support having the model be null. So, no need to initialize it anymore.

File: themis_components/thDatePicker/thDatePicker.directive.coffee [hunk 2]
```diff
@@ -1,15 +1,17 @@
 moment = require "moment"
 uuid = require "uuid"
 
-angular.module('ThemisComponents')
-  .directive 'thDatePicker', ->
+angular.module("ThemisComponents")
+  .directive "thDatePicker", ->
```
Comment (slapheadted) [inline themis_components/thDatePicker/thDatePicker.directive.coffee]: 🎉

File: themis_components/thFilter/examples/6 - Temporary Static Examples/coffee.coffee [hunk 3]
```diff
@@ -0,0 +1,22 @@
+angular.module 'thDemo', ['ThemisComponents']
```
Comment (slapheadted) [inline themis_components/thFilter/examples/6 - Temporary Static Examples/coffee.coffee]: Double-quotes :)

File: themis_components/thFilter/filters/date/dateFilter.service.coffee [hunk 4]
```diff
@@ -0,0 +1,20 @@
+moment = require "moment"
```
Comment (slapheadted) [inline themis_components/thFilter/filters/date/dateFilter.service.coffee]: Might need some tests to cover this service.

File: themis_components/thFilter/filters/date/thFilter.date.directive.coffee [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+    controller: ($scope) ->
+      @onValueChange = (event) =>
+        @filterSet.onFilterChange()
+
+      $scope.$on "$destroy", =>
+        @filterSet.remove @filter
+
+        if @filter.getState()?
+          @filterSet.onFilterChange()
+
```
Comment (slapheadted) [inline themis_components/thFilter/filters/date/thFilter.date.directive.coffee]: According our fancy new Style Guide, we would want to place a `return` at the end of a function (that isn't one simple statement). [:link:](https://github.com/slapheadted/themis/blob/1746f10a9217eb435e38bf37bf0a3de5d5d74192/dev-docs/coding_style_guide/coffeescript_style_guide.md#explicit-return)

File: themis_components/thFilter/filters/date/thFilter.date.directive.coffee [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+      return
+    compile: ->
+      pre: (scope, element, attrs, controller) ->
+        controller.filter = new DateFilter(
+          controller.filterOptions
+          controller.initialState
+        )
+
+        controller.filterSet.push controller.filter
```
Comment (slapheadted) [inline themis_components/thFilter/filters/date/thFilter.date.directive.coffee]: [:link:](https://github.com/slapheadted/themis/blob/1746f10a9217eb435e38bf37bf0a3de5d5d74192/dev-docs/coding_style_guide/coffeescript_style_guide.md#explicit-return)

File: themis_components/thFilter/filters/date/dateFilter.service.coffee (hunk 4 above)
Comment (JeffMarvin) [inline themis_components/thFilter/filters/date/dateFilter.service.coffee]: We will add tests as part of the cleanup/follow-up work shortly. I've got it tracked as tech debt in a list for processing after we're done.

File: themis_components/thFilter/filters/date/thFilter.date.directive.coffee (hunk 5 above)
Comment (JeffMarvin) [inline themis_components/thFilter/filters/date/thFilter.date.directive.coffee]: Thanks, that one will take me a bit to get used to. :)

File: themis_components/thFilter/examples/6 - Temporary Static Examples/coffee.coffee (hunk 3 above)
Comment (JeffMarvin) [inline themis_components/thFilter/examples/6 - Temporary Static Examples/coffee.coffee]: Corrected in this file.

File: themis_components/thFilter/filters/date/thFilter.date.directive.coffee (hunk 6 above)
Comment (JeffMarvin) [inline themis_components/thFilter/filters/date/thFilter.date.directive.coffee]: Added explicit return below.

File: themis_components/thFilter/examples/6 - Temporary Static Examples/coffee.coffee [hunk 7]
```diff
@@ -1,12 +1,12 @@
...
     FilterSet
   ) ->
     filterTypes = [
       {
-        name: 'Date'
-        type: 'date'
-        fieldIdentifier: 'exampleDate'
+        name: "Date"
+        type: "date"
+        fieldIdentifier: "exampleDate"
```
Comment (slapheadted) [inline themis_components/thFilter/examples/6 - Temporary Static Examples/coffee.coffee]: 👍

File: themis_components/thFilter/filters/date/thFilter.date.directive.coffee [hunk 8]
```diff
@@ -18,6 +18,8 @@ angular.module "ThemisComponents"
         if @filter.getState()?
           @filterSet.onFilterChange()
 
+        return
```
Comment (slapheadted) [inline themis_components/thFilter/filters/date/thFilter.date.directive.coffee]: 👍

File: themis_components/thFilter/filters/date/thFilter.date.directive.coffee [hunk 9]
```diff
@@ -30,3 +32,5 @@ angular.module "ThemisComponents"
         )
 
         controller.filterSet.push controller.filter
+
+        return
```
Comment (slapheadted) [inline themis_components/thFilter/filters/date/thFilter.date.directive.coffee]: 👍

**Pull request merged**